### PR TITLE
[CDAP-3188] [CDAP-3231] [CDAP-3236] [CDAP-3245] Fix/ui explore download queries

### PR DESCRIPTION
**Pull request by elfenheart**
- Fix downloading large query
- Fix missing description on created apps
- Fix nullable schema propagation
- Fix workflow schedules view when there is no schedule

### CODE REVIEW

File: cdap-ui/server/express.js [hunk 1]
```diff
@@ -60,6 +61,42 @@ function makeApp (authAddress, cdapConfig) {
...
+  app.post('/downloadQuery', function(req, res) {
+    var fs = require('fs');
+    var StringDecoder = require('string_decoder').StringDecoder;
+    var url = req.body.backendUrl;
+    var query = req.body.queryHandle;
+
+    var path = DIST_PATH + '/assets/public';
+
+    fs.mkdir(path, function (err) {
+      if (err && err.code === 'EEXIST') {
```
Comment (anwar6953) [inline cdap-ui/server/express.js]: doesn't seem like we need an error handling callback at all?
(since it's just a no-op)

Comment (elfenheart) [inline cdap-ui/server/express.js]: Unfortunately we do. If I don't specify the error callback, the node server crash

Comment (anwar6953) [inline cdap-ui/server/express.js]: ok

**Comment by ajainarayanan** [comment]
:+1: LGTM

**Pull request merged**